### ISSUE TITLE
Support embedding the Block Coding Editor in-game

### ISSUE THREAD

**Issue by sgaseretto** [endlessm/godot-block-coding#196]
### What Happened

This is not an issue but a question. Is it possible to embed the code editor in a game? Sorry if this is a very obvious question, I'm very new to godot and want to know if:
- The editor can be rendered insied a godot game
- We can create custom Blocks with specific functions, like move forward 5 times, turn left, move forward 5 times, and so on
- Can the generated code be executed while the game is running? (Don't know if gdscript is compiled at some point and cannot be executed dynamically while the game runs)

I know it sounds very meta, but basically the idea is to build something like [blockly-games](https://blockly.games/maze?lang=en&level=7&&skin=0)

**Comment by manuq** [comment]
@sgaseretto this would be interesting to investigate. The way it works today: the BlockCode node replaces the script of the parent node with the code generated by its blocks. So it wouldn't be too hard.

**Comment by parencompany** [comment]
Hope this happens!

This is just about 'killer feature' territory for me -- 
And can actually imagine way more usecases (personally), 
in-instance / in-game than using in-built in the existing editor.

Would be cool if there was like a `custom.gd` or something 
that we could just have a bunch of dict definitions for 
basic structure of block, category, etc (no idea how it works now lol)

**Comment by DavidGasku** [comment]
After looking at the code, seems like a big refactor to me. The whole code generation should be replaced by a interpreter, that runs one block each time.

**Comment by manuq** [comment]
@goatchurchprime awesome, this is what @dylanmccall was trying to do in this draft PR https://github.com/endlessm/godot-block-coding/pull/223 . Feel free to open a PR based on it.

**Comment by goatchurchprime** [comment]
Okay, I've done what I can to make it do the right thing, more or less
 https://github.com/endlessm/godot-block-coding/pull/280

**Comment by manuq** [comment]
I checked how far are we from embedding blocks in a game. Not far at all:
https://github.com/user-attachments/assets/4106b626-d45e-4277-8fae-483373b4200b

**Comment by goatchurchprime** [comment]
It's missing the palette of those components on the left hand side so far.  But I can seed the panel with various things before I run it, so this is workable in terms of proving the concept.  I'll try and get this into a multiplayer VR situation, in which case it might make sense for the palette panel to be in a different window.

**Comment by manuq** [comment]
> It's missing the palette of those components on the left hand side so far. But I can seed the panel with various things before I run it, so this is workable in terms of proving the concept. I'll try and get this into a multiplayer VR situation, in which case it might make sense for the palette panel to be in a different window.

Yes, the palette is missing on purpose. I was thinking that for in-game usage, the developer might be able to leave the palette out, or customize it with a set of blocks. I will try to move this draft PR forward soon.

**Comment by goatchurchprime** [comment]
I couldn't get it to work on my VR headset, which is based on Android.  After chasing it down for a long while I determined that it is a general problem to do with that `EditorInterface` Singleton on Android export, and reported it on the main repo as:  https://github.com/godotengine/godot/issues/103378

Not having worked with editor plugins, I don't have a clear idea about what is going on.  None of the other examples (eg pong) deploy to Android without encountering this error, so it's not just due to the in-game editing idea.  (Don't forget to Enable Remote Debugging.)  I haven't checked if there is a similar problem with other exports, but I wouldn't be surprised if the export templates are systematically cleansed of all references to the EditorInterface.

It could be that they were expecting addons to be either an editor plugin only, or nothing to do with the Editor interface, whereas this app is something halfway in between.  There will be a work-around, but it will require some thinking.

**Comment by manuq** [comment]
The demo in #380 should be anough to close this issue.
